syntax = "proto3";

option objc_class_prefix = "RTG";

package file;

// The greeting service definition.
service File {
  // Sends a greeting
  rpc PerUploadFile (PerFileRequest) returns (Token) {}

  rpc UploadFile (FileRequest) returns (FileReply) {}
}

// The request message containing the user's name.
message PerFileRequest {
  int32 fileByte = 1;
  int64 num = 2;
  string str = 3;
}

// The response message containing the greetings
message Token {
  uint64 hash = 1;
}

message FileRequest {
  bytes content =1;
}

message FileReply{
  string reply = 1;
}
